pkg/urbit/noun/manage.c

```diff
@@ -1633,6 +1634,11 @@ u3m_boot(c3_c* dir_c)
   */
   u3m_init();
 
+  /* In the worker, set the openssl memory allocation functions to always
+  ** work on the loom.
+  */
+  CRYPTO_set_mem_functions(u3a_malloc, u3a_realloc, u3a_free);
```

Comment:
I'm really surprised this worked. From the OpenSSL docs:

```
 int CRYPTO_set_mem_functions(
         void *(*m)(size_t, const char *, int),
         void *(*r)(void *, size_t, const char *, int),
         void (*f)(void *, const char *, int))
```


And from the build output:

```
noun/manage.c: In function 'u3m_boot':
noun/manage.c:1640:28: warning: passing argument 1 of 'CRYPTO_set_mem_functions' from incompatible pointer type [-Wincompatible-pointer-types]
   CRYPTO_set_mem_functions(u3a_malloc, u3a_realloc, u3a_free);
                            ^~~~~~~~~~
In file included from noun/manage.c:10:0:
/nix/store/nx7qi0av2h4bfw76fakm6b8a73g27vzx-openssl-1.1.1-x86_64-linux-musleabi/include/openssl/crypto.h:261:5: note: expected 'void * (*)(size_t,  const char *, int) {aka void * (*)(long unsigned int,  const char *, int)}' but argument is of type 'void * (*)(size_t) {aka void * (*)(long unsigned int)}'
 int CRYPTO_set_mem_functions(
     ^~~~~~~~~~~~~~~~~~~~~~~~
noun/manage.c:1640:40: warning: passing argument 2 of 'CRYPTO_set_mem_functions' from incompatible pointer type [-Wincompatible-pointer-types]
   CRYPTO_set_mem_functions(u3a_malloc, u3a_realloc, u3a_free);
                                        ^~~~~~~~~~~
In file included from noun/manage.c:10:0:
/nix/store/nx7qi0av2h4bfw76fakm6b8a73g27vzx-openssl-1.1.1-x86_64-linux-musleabi/include/openssl/crypto.h:261:5: note: expected 'void * (*)(void *, size_t,  const char *, int) {aka void * (*)(void *, long unsigned int,  const char *, int)}' but argument is of type 'void * (*)(void *, size_t) {aka void * (*)(void *, long unsigned int)}'
 int CRYPTO_set_mem_functions(
     ^~~~~~~~~~~~~~~~~~~~~~~~
noun/manage.c:1640:53: warning: passing argument 3 of 'CRYPTO_set_mem_functions' from incompatible pointer type [-Wincompatible-pointer-types]
   CRYPTO_set_mem_functions(u3a_malloc, u3a_realloc, u3a_free);
                                                     ^~~~~~~~
In file included from noun/manage.c:10:0:
/nix/store/nx7qi0av2h4bfw76fakm6b8a73g27vzx-openssl-1.1.1-x86_64-linux-musleabi/include/openssl/crypto.h:261:5: note: expected 'void (*)(void *, const char *, int)' but argument is of type 'void (*)(void *)'
 int CRYPTO_set_mem_functions(
     ^~~~~~~~~~~~~~~~~~~~~~~~
```

The default functions are not OPENSSL_malloc/realloc/free, but CRYPTO_malloc/realloc_free.